browser/ui/views/toolbar/brave_vpn_button.cc

```diff
@@ -127,9 +131,12 @@ BraveVPNButton::BraveVPNButton(Browser* browser)
       browser_(browser),
       service_(brave_vpn::BraveVpnServiceFactory::GetForProfile(
           browser_->profile())) {
-  CHECK(service_);
-  UpdateButtonState();
-  Observe(service_);
+  if (service_) {
+    UpdateButtonState();
```

Comment:
where is this crashing?

Reply:
it does not look like this service should be null in testing, it should only be null when it's not allowed for the profile so I think this is a code issue, not a test issue

Reply:
yep, `BuildVpnService` also returns nullptr is `g_brave_browser_process->brave_vpn_connection_manager() ` nullptr so the checks are out of sync with each other so I think we should replace `brave_vpn::IsAllowedForContext` with `BraveVpnServiceFactory::GetForProfile() != nullptr` in BraveToolbarView

Reply:
Updated in 1d61e34a570